R/zchunk_L100.Population_downscale_ctry.R

```diff
@@ -33,54 +34,191 @@ module_socioeconomics_L100.Population_downscale_ctry_DISABLED <- function(comman
     SSP_database_v9 <- get_data(all_data, "socioeconomics/SSP_database_v9")
     UN_popTot <- get_data(all_data, "socioeconomics/UN_popTot")
 
+if(1) {
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    # ===================================================
 
+    ## (1) Historical population by country
+
+    # First clean up Maddison raw data -- NOTE: Maddison data are used to develop population ratios relative to 1950 to combine with UN data from 1950 onward
+    pop_thous_ctry_reg <- Maddison_population %>%
+      rename(Maddison_ctry = Country, pop = value) %>%  # Change name to match socioeconomics_ctry mapping file
+      left_join(socioeconomics_ctry, by = "Maddison_ctry") # Join with iso codes
```

Comment:
There are places in the file where we definitely want to keep the NAs and some joins where it doesn't matter if they are there or not. I just defaulted to the left_join because there are so many places where losing the NAs messes things up later. Let me know if you want me to change it.